pkg/services/govmomi/service.go

```diff
@@ -159,7 +159,7 @@ func (vms *VMService) DestroyVM(ctx *context.VMContext) (infrav1.VirtualMachine,
 	if err != nil {
 		// If the VM's MoRef could not be found then the VM no longer exists. This
 		// is the desired state.
-		if isNotFound(err) {
+		if isNotFound(err) || isFolderNotFound(err) {
```

Comment:
Do you want to try using https://golang.org/pkg/errors/#Is ?

Reply:
Or https://godoc.org/github.com/pkg/errors#Is given that's the package we actually use

Reply:
@vincepri - I like this. can you raise an issue for this, I'd like to refactor the error handling to use it ?

Reply:
#1004